packages/target-tauri/src-tauri/src/html_window/commands.rs

```diff
@@ -0,0 +1,167 @@
+        HTML_EMAIL_WARNING_DEFAULT,
+    );
+
+    if warning && load_remote_content {
+        let (tx, rx) = tokio::sync::oneshot::channel::<bool>();
+
+        app.dialog()
+            .message("tx('load_remote_content_ask')")
+            .parent(&webview.window())
+            .buttons(tauri_plugin_dialog::MessageDialogButtons::OkCancelCustom(
+                // TODO use translation strings, as soon as translations are avaialble in rust backend
+                "Yes".to_owned(),
+                "No".to_owned(),
+            ))
+            .show(|answer| tx.send(answer).unwrap());
+        if !rx.await? {
+            return Err(Error::UserCanceled);
+        }
+    }
+
+    set_load_remote_content(&webview, load_remote_content).await
+}
+
+#[tauri::command]
+pub(crate) fn html_email_open_menu(
+    app: tauri::AppHandle,
+    webview: tauri::Webview,
+    html_instances_state: tauri::State<HtmlEmailInstancesState>,
+) -> Result<(), Error> {
+    let desktop_settings = app.store(CONFIG_FILE)?;
+    let always_load = get_setting_bool_or(
+        desktop_settings.get(HTML_EMAIL_ALWAYS_ALLOW_REMOTE_CONTENT_KEY),
+        HTML_EMAIL_ALWAYS_ALLOW_REMOTE_CONTENT_DEFAULT,
+    );
+    let warning = get_setting_bool_or(
+        desktop_settings.get(HTML_EMAIL_WARNING_KEY),
+        HTML_EMAIL_WARNING_DEFAULT,
+    );
+
+    let instance_state = block_on(html_instances_state.get(webview.window().label()))
+        .ok_or(Error::WindowNotFoundInState)?;
+
+    let always_load_remote_images = CheckMenuItem::new(
+        &app,
+        "tx('always_load_remote_images')",
+        true,
+        always_load,
+        None::<&str>,
+    )?;
+    let show_warning = CheckMenuItem::new(&app, "tx('show_warning')", true, warning, None::<&str>)?;
+
+    let menu = if instance_state.is_contact_request {
+        Menu::with_items(&app, &[&show_warning])
+    } else {
+        Menu::with_items(&app, &[&always_load_remote_images, &show_warning])
+    }?;
+
+    // treefit: I haven't found an easy way to get the title bar offset,
+    // so now we just spawn it at mouse position
+    webview.window().popup_menu(&menu)?;
+    webview.window().on_menu_event(move |window, event| {
+        // IDEA: better error handling
+        let desktop_settings = window.store(CONFIG_FILE).expect("config file not found");
+        if event.id() == always_load_remote_images.id() {
+            let new_always_load = !always_load;
+            desktop_settings.set(HTML_EMAIL_ALWAYS_ALLOW_REMOTE_CONTENT_KEY, new_always_load);
+            block_on(set_load_remote_content(&webview, new_always_load)).unwrap();
+
+            // TODO: implement reload in tauri
+            webview.eval("location.reload()").unwrap();
+        }
+        if event.id() == show_warning.id() {
+            desktop_settings.set(HTML_EMAIL_WARNING_KEY, !warning);
+        }
+    });
+    Ok(())
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
```

Comment:
Does this mean all properties/keys of the struct will be camel cased when serialized?

Reply:
yes, https://serde.rs/container-attrs.html#rename_all